### PR TITLE
Ingest failed transactions

### LINKED ISSUE #309: transactions that fail during consensus are ignored by Horizon
See details https://github.com/stellar/stellar-core/issues/1541

Net is:
* a failed transaction is not recorded by Horizon (lookup by tx hash doesn't return anything)
* the count of transactions reported as part of the `ledgers` endpoint is reporting the number of ingested transactions and the data attached to the transactions doesn't match what is recorded in core's history.

In the issue linked above, the ledger contained 3 transactions (not 2 as reported by the endpoint), and the tx indexes are wrong (as in, the first transaction reported is actually not the first transaction of the transaction set).

This may explain https://github.com/stellar/go/issues/305 as clients will timeout instead of getting notified of the failure.

### PR DESCRIPTION
**Pull request by bartekn**
TODO:
- [x] Config feature flag to enable ingesting failed transactions (looks like it can double the space requirement for a DB).
- [x] Ignore failed txs during assets stats calculations.
- [x] `scc` updates to allow transaction failures: https://github.com/stellar/stellar_core_commander/pull/128
- [x] Rebuild test scenarios (after review for sanity).
- [ ] SDKs updates.

Close #309.

### CODE REVIEW

File: services/horizon/internal/db2/history/operation.go [hunk 1]
```diff
@@ -122,6 +122,13 @@ func (q *OperationsQ) OnlyPayments() *OperationsQ {
 	return q
 }
 
+// IncludeFailed changes the query to include failed transactions.
```
Comment (tomerweller) [inline services/horizon/internal/db2/history/operation.go]: documentation doesn't match function

File: services/horizon/internal/db2/history/transaction.go [hunk 2]
```diff
@@ -72,6 +72,13 @@ func (q *TransactionsQ) ForLedger(seq int32) *TransactionsQ {
 	return q
 }
 
+// IncludeFailed changes the query to include failed transactions.
```
Comment (tomerweller) [inline services/horizon/internal/db2/history/transaction.go]: documentation doesn't match function

**Comment by bartekn** [comment]
> I wonder if the concept of "failed operations" will ever come back to haunt us. An Operation can have a successful `OperationResult` but still not be applied due to being part of a failed transaction, and be called "failed". Maybe it's just semantics. 

This is a great point. Not sure if it's still the case but I think in the past I've seen this extreme case when there was a problem with transaction itself and all ops results were set to success by default. Anyway, I'll think about a solution. The easiest one seems to be renaming `successful` to `transaction_successful` in operation resource.

**Comment by bartekn** [comment]
@tomerweller can you PTAL again? In "Files changed" view it's possible to filter out `.sql` files and the diff is actually quite readable.

File: services/horizon/internal/actions_operation_fee_stats.go [hunk 3]
```diff
@@ -26,6 +28,20 @@ type OperationFeeStatsAction struct {
...
+			Type:   "endpoint_not_available",
+			Title:  "Endpoint Not Available",
+			Status: http.StatusNotImplemented,
+			Detail: "/operation_fee_stats is unavailable when Horizon is not ingesting failed " +
+				"transactions. Set `INGEST_FAILED_TRANSACTIONS=true` to start ingesting them.",
+		}
+		problem.Render(action.R.Context(), action.W, p)
+		return nil
+	}
+
```
Comment (tomerweller) [inline services/horizon/internal/actions_operation_fee_stats.go]: Rethinking this. What do you think about still serving this endpoint but omit the statistical data if failed transactions are not ingested? the `last_ledger_base_fee` should still be available and might be used by sdks as a basic fee strategy  (https://github.com/stellar/js-stellar-sdk/issues/233).

Comment (bartekn) [reply services/horizon/internal/actions_operation_fee_stats.go]: Your idea is good, but I'm thinking if people will always remember to check if other values are available. What do you think about getting `last_ledger_base_fee` by querying `/ledgers?order=desc&limit1`?

I was also thinking about ingesting failed transactions in recent ledgers by default and removing them periodically when `INGEST_FAILED_TRANSACTIONS=false`. But it's much more complicated code and we won't be able to ship it this week.

Comment (tomerweller) [reply services/horizon/internal/actions_operation_fee_stats.go]: Ok I'm ok with querying last ledger

**Pull request merged**